libs/remix-simulator/src/vm-context.ts

```diff
@@ -124,6 +88,158 @@ class StateManagerCommonStorageDump extends DefaultStateManager {
   }
 }
 
+export interface CustomEthersStateManagerOpts {
+  provider: string | ethers.providers.StaticJsonRpcProvider | ethers.providers.JsonRpcProvider
+  blockTag: bigint | 'earliest',
+  /**
+   * A {@link Trie} instance
+   */
+  trie?: Trie
+}
+
+class CustomEthersStateManager extends StateManagerCommonStorageDump {
+  private provider: ethers.providers.StaticJsonRpcProvider | ethers.providers.JsonRpcProvider
+  private blockTag: string
+
+  constructor(opts: CustomEthersStateManagerOpts) {
+    super(opts)
+    if (typeof opts.provider === 'string') {
+      this.provider = new ethers.providers.StaticJsonRpcProvider(opts.provider)
+    } else if (opts.provider instanceof ethers.providers.JsonRpcProvider) {
+      this.provider = opts.provider
+    } else {
+      throw new Error(`valid JsonRpcProvider or url required; got ${opts.provider}`)
```

Comment:
`provider` can be of 3 types `string | ethers.providers.StaticJsonRpcProvider | ethers.providers.JsonRpcProvider`. Shouldn't there be a third check for `else if (opts.provider instanceof ethers.providers.StaticJsonRpcProvider)` before the else statement?